src/main/java/com/kth/ciserver/controller/CiController.java

```diff
@@ -25,6 +26,34 @@ public String handleGithubWebhook(@RequestBody GithubWebhookRequest request) {
     }
 
     private void pullAndBuildApplication(GithubWebhookRequest request) {
+        executeAndPrintCommand("git pull");
+        executeAndPrintCommand(String.format("git checkout %s", request.getHead_commit().getId()));
+        Boolean testsSuccessful = executeAndPrintCommand("./gradlew clean build");
+        System.out.println("Tests successful: " + testsSuccessful); // this should instead be printed to file for consistent storage
+    }
 
+    
+    /** 
+     * @param cmd Command to be executed and printed
+     * @return True if the second word in the second to last line of the standard output 
+     * is "SUCCESSFUL" (indicating a successful test run), else false.
+     */
+    private Boolean executeAndPrintCommand(String cmd) {
+        ArrayList<String> lines = new ArrayList<>();
+        try {
+            Runtime run = Runtime.getRuntime();
+            Process process = run.exec(cmd);
+            BufferedReader buf = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line;
+            while ((line = buf.readLine()) != null) {
+                System.out.println(line);
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
```

Comment:
This is a good idea to print the IOException. But I think we should carry on throwing the IOException up the chain of methods, so that it stops the process since something went wrong. E.g if the Git pull breaks for somereason, we should stop, and shouldn't carry on trying to run the pullAndBuildApplication